wgengine/wgengine.go

```diff
@@ -45,31 +58,55 @@ func (rs *RouteSettings) OnlyRelevantParts() string {
 		rs.LocalAddr, rs.DNS, rs.DNSDomains, peers)
 }
 
+// Router is the TODO.
 type Router interface {
+	// Up brings the router up.
+	// TODO: more than once? after Close?
 	Up() error
-	SetRoutes(rs RouteSettings) error
+	// SetRoutes sets the routes.
+	// TODO: while running?
+	SetRoutes(RouteSettings) error
+	// Close closes the router.
+	// TODO: return an error? does this block?
```

Comment:
It does not block on network I/O. (I don't think it works properly at all, but if it did it still wouldn't block.) Returning an error is probably a good idea.